pycloudlib/azure/cloud.py

```diff
@@ -863,6 +957,17 @@ def launch(
         # resources we are creating.
         self.tag = get_timestamped_tag(self.base_tag)
 
+        try:
+            self._check_compute_quota(instance_type)
+
+        except Exception as e:
+            # Clean up resources created during the failed attempt
+            self._log.error("Exception during launch: %s", e)
+            if self.resource_group:
+                self.delete_resource_group(self.resource_group.name)
+                self.resource_group = None
+            raise
```

Comment:
get rid of this cleanup for now. if cleanup errors persist, let's open a new issue and fix in a separate PR.